### PR TITLE
feat(account-service):#483 implement ticket update logic  

### LINKED ISSUE #483: [TICKETING] [IMPLEMENTATION] Implement update logic and data persistence
### Description
Finalize the ticket update flow by implementing the functional logic within the account-service.

In this final step, the TicketController will be fully integrated with the TicketRepository to perform the update operation. This includes retrieving the existing entity, mapping the fields from the update DTO, and persisting the changes back to the database.

### Acceptance Criteria

- [ ] The TicketController successfully calls the TicketRepository to update ticket data.
- [ ] The logic correctly maps allowed fields (title, description) from the request to the existing Ticket entity.
- [ ] Changes are successfully persisted in the database.
- [ ] The implementation includes error handling for cases where the ticket ID is not found.
- [ ] Integration tests verify the full "update and save" cycle, ensuring the data is correctly modified in storage.

### PR DESCRIPTION
**Pull request by jrffruit518-enrich**
Summary
This PR implements the Happy Path for the Ticket Update flow within the account-service. It focuses on the core functional logic to enable successful ticket updates for authenticated users.

Controller Implementation: Implemented the update endpoint in TicketController using @AuthenticationPrincipal OAuth2User to securely retrieve the user identity directly from the session.

Domain Integrity: Utilizes the Ticket.restore() method to ensure that core fields remain consistent and protected during the update process.

Persistence: Successfully connects the web adapter to the TicketRepository.updateTicket() method, ensuring data is correctly persisted in the InMemoryTicketRepository.

Testing: Includes targeted integration tests using MockMvc and oauth2Login() to verify the successful update scenario (200 OK).

Notes & Dependencies
Atomic PR Strategy: Following the project's quality guidelines, this PR focuses strictly on the success path. Error handling (401, 403) and corresponding edge-case tests will be submitted in a subsequent follow-up PR to maintain a clean and reviewable history.

Save Implementation: The save logic included in the repository and tests is for verification purposes of the update flow. The official save flow implementation is managed in a separate task.

part of  #483

### CODE REVIEW

**Review by JordiCasas87 — APPROVED** [review]
Looks good to me. Focused scope, clear ownership validation flow and good coverage for 403/404 scenarios.

The dependency on #657 and the temporary save implementation are also clearly documented.

**Review by eariasvalor — CHANGES_REQUESTED** [review]
Hi Rong, "gan de hao!" 
I have only one suggestion: the description says that tests verify unauthorized access (403), but there is no test to assess that scenario, could you please add it to the TicketControllerTest?

**Comment by jrffruit518-enrich** [comment]
> Hi Rong, "gan de hao!" I have only one suggestion: the description says that tests verify unauthorized access (403), but there is no test to assess that scenario, could you please add it to the TicketControllerTest?

Thank you for the feedback! You're right, I missed that one. I've just added the test case for the 403 Forbidden scenario to ensure the ownership validation is fully covered. Thanks for catching that!

**Review by andyrouge81 — APPROVED** [review]
The update flow and ownership validation look coherent and aligned with the current MVP approach.

Just as a small suggestion, since the PR is becoming a bit large mostly because of the tests, maybe you could consider removing the repository test `should_update_ticket_data_successfully` to save a few lines and keep the scope even more lightweight for this iteration.

The main update flow already seems well covered through the controller tests and the HTTP cases (`200`, `403`, `404`), so this repository test could probably be added later without affecting the MVP.

Apart from that, the implementation looks good to me. GJ!

**Review by Andrescalvo22 — APPROVED** [review]
The code looks good, just maybe for the next time, trie to make as few changes as possible in each PR!

**Review by eariasvalor — APPROVED** [review]
Thanks for the changes! Approved.

File: backend/account-service/src/test/java/com/ita/challenges/account/infrastructure/adapter/web/in/TicketControllerTest.java [hunk 1]
```diff
@@ -49,22 +49,59 @@ void should_return_200_with_ticket_list_when_requesting_ticket_list() throws Exc
...
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(updateRequest)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(ticketId))
+                .andExpect(jsonPath("$.title").value("New Title"))
+                .andExpect(jsonPath("$.userId").value(currentUserId));
+    }
+
+    @Test
+    void should_return_404_when_updating_non_existing_ticket() throws Exception {
```
Comment (kenan-it-academy) [inline backend/account-service/src/test/java/com/ita/challenges/account/infrastructure/adapter/web/in/TicketControllerTest.java]: Error cases (both tests and implementation) in a separate PR

File: backend/account-service/src/test/java/com/ita/challenges/account/infrastructure/adapter/web/in/TicketControllerTest.java [hunk 2]
```diff
@@ -49,22 +49,59 @@ void should_return_200_with_ticket_list_when_requesting_ticket_list() throws Exc
...
+                        .with(csrf())
+                        .with(oauth2Login().attributes(attrs -> attrs.put("login", "any-user")))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(updateRequest)))
+                .andExpect(status().isNotFound());
+    }
     @Test
-    @WithMockUser
-    void should_return_error_when_updating_ticket_because_not_implemented() throws Exception {
+    void should_return_403_when_user_is_not_the_owner() throws Exception {
```
Comment (kenan-it-academy) [inline backend/account-service/src/test/java/com/ita/challenges/account/infrastructure/adapter/web/in/TicketControllerTest.java]: Error cases (both tests and implementation) in a separate PR

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]
Doesn't look bad, but please split happy path from error case PRs

**Comment by jrffruit518-enrich** [comment]
> Doesn't look bad, but please split happy path from error case PRs

Understood, Kenan. I have split the implementation as requested: PR #483 is now strictly for the happy path, and I have opened a new chained PR for the error cases and security logic.

**Review by kenan-it-academy — APPROVED** [review]
